File: Selenium-javascript/tests/network-mock.test.js
const {Builder, By} = require('selenium-webdriver');
const path = require('path');
const assert = require('assert');

// TODO: Intercept and mock the API response according to CandidateEvaluation.md

describe('mock API response', function() {
  this.timeout(30000);
  let driver;

  before(async () => {
    driver = new Builder().forBrowser('chrome').build();
  });

  after(async () => {
    if (driver) {
      await driver.quit();
    }
  });

  it('displays mocked data', async () => {
    const filePath = path.resolve(__dirname, '../../public/api.html');
    await driver.get('file://' + filePath);

    // Use browser devtools protocol or proxy to mock the request
    // await driver.findElement(By.id('load-user')).click();
    // const output = await driver.findElement(By.id('output')).getText();
    // assert(output.includes('Jane Doe'));
  });
});

File: Selenium-java/src/test/java/NetworkMockTest.java
import org.junit.jupiter.api.*;
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import java.nio.file.Paths;

public class NetworkMockTest {
    private WebDriver driver;

    @BeforeEach
    public void setUp() {
        driver = new ChromeDriver();
    }

    @AfterEach
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }

    @Test
    public void mockApiResponse() {
        String fileUrl = Paths.get("../public/api.html").toUri().toString();
        driver.get(fileUrl);

        // TODO: Intercept the network request to https://randomuser.me/api/
        // and return a mocked response using a proxy or CDP commands.

        // driver.findElement(By.id("load-user")).click();
        // WebElement output = driver.findElement(By.id("output"));
        // Assertions.assertTrue(output.getText().contains("Jane Doe"));
    }
}
